mythril/laser/ethereum/svm.py

```diff
@@ -180,6 +171,22 @@ def sym_exec(
             )
             log.info("Achieved {:.2f}% coverage for code: {}".format(cov, code))
 
+    def sym_exec(
+        self, main_address=None, creation_code=None, contract_name=None, timeout=None
+    ) -> None:
+        timeout = timeout or self.execution_timeout
+        with ThreadPoolExecutor(max_workers=1) as executor:
+            f_results = executor.submit(
+                self._sym_exec,
+                main_address=main_address,
+                creation_code=creation_code,
+                contract_name=contract_name,
+            )
+            try:
+                return f_results.result(timeout)
+            except TimeoutError:
+                return
```

Comment:
In `rubixi.sol`, as @norhh pointed out, execution still hangs. It turns out that the `TimeoutError` is being raised and caught, but execution gets stuck returning, presumably waiting patiently for the `with` to gracefully terminate `executor`. We might need to kill the thread a little more aggressively.

Reply:
You are right, unfortunately Executors & futures don't allow this so I used ThreadPool instead